libsupport/include/katana/ArrowVisitor.h

```diff
@@ -207,18 +212,31 @@ std::enable_if_t<
         internal::is_visit_arrow_base_t<Arg0>,
         internal::is_visit_arrow_base_t<Args>...>,
     katana::Result<typename std::decay_t<VisitorType>::ReturnType>>
-VisitArrow(VisitorType&& visitor, Arg0 array, Args... args) {
-  return internal::Wrapper<>::template VisitArrowInternal<
+VisitArrow(VisitorType&& visitor, const Arg0& arg0, const Args&... args) {
+  return internal::VisitArrowInternalWrapper<>::template VisitArrowInternal<
       internal::ArrayVisitorBaseType>(
       std::forward<VisitorType>(visitor), std::tuple<>{},
-      std::forward<Arg0>(array), std::forward<Args>(args)...);
+      std::forward<Arg0>(arg0), std::forward<Args>(args)...);
+}
+
+// VisitArrow call that supports shared_ptr args for the same
+// types as above
+template <class VisitorType, class Arg0, class... Args>
+std::enable_if_t<
+    std::conjunction_v<
+        internal::is_visit_arrow_base_sp_t<Arg0>,
+        internal::is_visit_arrow_base_sp_t<Args>...>,
+    katana::Result<typename std::decay_t<VisitorType>::ReturnType>>
+VisitArrow(VisitorType&& visitor, const Arg0& arg0, const Args&... args) {
+  return VisitArrow(
+      std::forward<VisitorType>(visitor), *(arg0.get()), (*(args.get()))...);
```

Comment:
FWIW, *ptr <=> *ptr.get() where ptr is shared_ptr.

Reply:
yeah I originally thought that *ptr.get() might give me the underlying data type instead of the shared_ptr wrapper which would allow me to avoid using decay types. Fogot to change it back after I figured that out so I'll do it now.